vumi/persist/tests/test_fake_redis.py

```diff
@@ -178,6 +186,14 @@ def test_zremrangebyrank_negative_stop(self):
             [('one', 1)], 'zrange', 'set', 0, -1, withscores=True)
 
     @inlineCallbacks
+    def test_zremrangebyrank_negative_stop_empty_range(self):
+        yield self.redis.zadd('set', one=1, two=2, three=3)
+        yield self.assert_redis_op(0, 'zremrangebyrank', 'set', 0, -5)
```

Comment:
Ah, right. The problem here was that we were mishandling all negative values, so a single test case each for the combinations of empty and non-empty ranges with a negative value at each end is sufficient.